### PR TITLE
Mintlayer wallet implementation

### PR DESCRIPTION
**Pull request by YunkevichAndrei**

### CODE REVIEW

**Review by anyxem — CHANGES_REQUESTED** [review]
components/amoint_input/AmountInput.js
looks like there is typo in folder name, should be amount_input

File: tests/integration/Currency.test.js [hunk 1]
```diff
@@ -14,25 +14,26 @@ describe('currency', () => {
     assert.ok(cur[currency.LAST_UPDATED] > 0);
     assert.ok(cur.BTC_USD > 0);
 
+    // disabled, because Mintlayer currency API doesn't currently support other fiat units
```
Comment (anyxem) [inline tests/integration/Currency.test.js:17]: We have separate API for ml currency rate
http://51.158.172.176:3000/getCurrentRate/ml/usd

**Review by anyxem — CHANGES_REQUESTED** [review]

File: screen/send/details/hooks/useCreateTransaction.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    let recipients = outputs.filter(({ address }) => address !== changeAddress);
+
+    if (recipients.length === 0) {
+      // special case. maybe the only destination in this transaction is our own change address..?
+      // (ez can be the case for single-address wallet when doing self-payment for consolidation)
+      recipients = outputs;
+    }
+
+    navigation.navigate('Confirm', {
+      fee: new BigNumber(fee).dividedBy(100000000).toNumber(),
```
Comment (anyxem) [inline screen/send/details/hooks/useCreateTransaction.js:139]: 100000000 might be some const

**Review by anyxem — CHANGES_REQUESTED** [review]

File: screen/send/details/hooks/useRecalcFee.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+          } else if (transaction.amount) {
+            if (currency.btcToSatoshi(transaction.amount) > 0) {
+              targets.push({ address: transaction.address, value: currency.btcToSatoshi(transaction.amount) });
+            }
+          }
+        }
+
+        // if targets is empty, insert dust
+        if (targets.length === 0) {
+          targets.push({ address: '36JxaUrpDzkEerkTf1FzwHNE1Hb7cCjgJV', value: 546 });
```
Comment (anyxem) [inline screen/send/details/hooks/useRecalcFee.js:46]: What is that addresses? Is that refactoring from the bluewallet part?

File: blue_modules/mintlayer/mintlayer.js [hunk 4]
```diff
@@ -61,14 +62,14 @@ export const getTxInput = async (outpointSourceId, index) => {
   return webviewEventBus.exec(wasmMethods.encode_input_for_utxo, [outpointSourceId, index]);
 };
 
-export const getOutputs = async (amount, address, networkType) => {
+export const getOutputs = async ({ amount, address, networkType }) => {
```
Comment (anyxem) [inline blue_modules/mintlayer/mintlayer.js:65]: getOutputs is more complex, refer to 
https://github.com/mintlayer/mojito-browser-extension/blob/main/src/services/Crypto/Mintlayer/Mintlayer.js#L166

**Review by anyxem — CHANGES_REQUESTED** [review]

File: blue_modules/Mintlayer.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
+const prefix = '/api/v1';
+
+const MINTLAYER_ENDPOINTS = {
+  GET_ADDRESS_DATA: '/address/:address',
+  GET_TRANSACTION_DATA: '/transaction/:txid',
+  GET_ADDRESS_UTXO: '/address/:address/available-utxos',
```
Comment (anyxem) [inline blue_modules/Mintlayer.js:6]: this endpoint is deprecated, use
```js
  GET_ADDRESS_UTXO: '/address/:address/spendable-utxos',
```

**Comment by anyxem** [comment]
In scope of this PR upgrade api to v2

File: screen/send/details/hooks/useRecalcFee.js (hunk 3 above)
Comment (YunkevichAndrei) [reply screen/send/details/hooks/useRecalcFee.js:46]: Right, It's refactoring from bluewallet part, I just moved it to hook.

File: screen/send/details/hooks/useCreateTransaction.js (hunk 2 above)
Comment (YunkevichAndrei) [reply screen/send/details/hooks/useCreateTransaction.js:139]: It's refactoring from bluewallet part, I just moved it to hook.

File: blue_modules/Mintlayer.js (hunk 5 above)
Comment (YunkevichAndrei) [reply blue_modules/Mintlayer.js]: Done and migrated to api v2.

File: blue_modules/mintlayer/mintlayer.js (hunk 4 above)
Comment (YunkevichAndrei) [reply blue_modules/mintlayer/mintlayer.js]: Done

File: tests/integration/Currency.test.js (hunk 1 above)
Comment (YunkevichAndrei) [reply tests/integration/Currency.test.js:17]: Correct, that's why it's disabled temporary, because this API doesn't support other fiat units.

**Pull request merged**